### DISCUSSION TITLE
Unable to start codelldb (exit code 1)

### DISCUSSION THREAD

**Discussion by asmodeus812** [mfussenegger/nvim-dap#492] [General]
### Debug adapter definition and debug configuration

I downloaded the codelldb binaries from here https://github.com/vadimcn/vscode-lldb/releases/latest/download/codelldb-x86_64-linux.vsix and did unzip CodeLLDB.zip. Then configured nvim dap as specified in the main page using the following configurations
`
 nvim_dap.adapters.codelldb = function(on_adapter)
            local stdout = vim.loop.new_pipe(false)
            local stderr = vim.loop.new_pipe(false)

            local cmd = '/home/user/.local/share/nvim/lsp_dbgadapters/codelldb/extension/adapter/codelldb'

            local handle, pid_or_err
            local opts = {
                stdio = { nil, stdout, stderr },
                detached = true,
            }
            handle, pid_or_err = vim.loop.spawn(cmd, opts, function(code)
                stdout:close()
                stderr:close()
                handle:close()
                if code ~= 0 then
                    print("codelldb exited with code", code)
                end
            end)
            assert(handle, "Error running codelldb: " .. tostring(pid_or_err))
            stdout:read_start(function(err, chunk)
                assert(not err, err)
                if chunk then
                    local port = chunk:match('Listening on port (%d+)')
                    if port then
                        vim.schedule(function()
                            on_adapter({
                                type = 'server',
                                host = '127.0.0.1',
                                port = port
                            })
                        end)
                    else
                        vim.schedule(function()
                            require("dap.repl").append(chunk)
                        end)
                    end
                end
            end)
            stderr:read_start(function(err, chunk)
                assert(not err, err)
                if chunk then
                    vim.schedule(function()
                        require("dap.repl").append(chunk)
                    end)
                end
            end)
        end,
        nvim_dap.configurations.c = {
            {
                name = "Launch",
                type = "codelldb",
                request = "launch",
                program = function()
                    return vim.fn.input("Path to executable: ", vim.fn.getcwd() .. "/", "file")
                end,
                cwd = vim.fn.getcwd(),
                stopOnEntry = false,
            },
        },
`

### Debug adapter version

Latest

### Steps to Reproduce

1. Setup nvim dap
2. Build a simple project in debug with cmake
3. Open the project 
4. = optional put a breakpoint somewhere =
5. Do lua require('dap').continue()
6. echo area prints "codelldb exited with code 1"

### Expected Result

The application is executed and entrypoint triggered.

### Actual Result

I receive error message codelldb exited with code 1, tirggered from the following line in the configuration itslef -> print("codelldb exited with code", code)

**Comment by mfussenegger** [comment]
You redirect the stderr output to the repl. Is there any output in it? 

Moving this to discussion because its a configuration issue and unlikely a client bug

**Reply by asmodeus812** [reply]
@mfussenegger I did manage to make it log to a file and this is what i get - Error: "Either --connect or --port must be specified. With the config as specified in the repo/readme guide for codelldb. Also the chunk from stdout:read_start(function(err, chunk) callback is NULL for some reason, and probably why it does not even reach the port/host part.

**Reply by mfussenegger** [reply]
Looks like https://github.com/vadimcn/vscode-lldb/commit/08dcebab314acd3ab275e6af23c8b867e6548135 removed the support for not providing any option and implicitly listening to some free port.

You now need to provide the `--port` option with a pre-defined port.
